crossplane/crossplane-secrets.yaml

```diff
@@ -0,0 +1,52 @@
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  annotations:
+    argocd.argoproj.io/sync-wave: "-2"
+  name: crossplane-secrets
+  namespace: crossplane-system
+spec:
+  dataFrom:
+    - extract:
+        conversionStrategy: Default
+        decodingStrategy: None
+        key: /crossplane
+  refreshInterval: 10s
+  secretStoreRef:
+    kind: ClusterSecretStore
+    name: vault-kv-secret
+  target:
+    creationPolicy: Owner
+    deletionPolicy: Retain
+    name: crossplane-secrets
+---
+apiVersion: external-secrets.io/v1beta1
+kind: ExternalSecret
+metadata:
+  name: git-credentials
+  annotations:
+    argocd.argoproj.io/sync-wave: "-2"
+spec:
+  target:
+    name: git-credentials
+    template:
+      engineVersion: v2
+      data:
+        creds: |
+          https://{{ .username }}:{{ .password }}@github.com
```

Comment:
I am not sure if this method is allowed for all organization (the basic auth), is not better to use PAT?

Reply:
this spec was borrowed from the approach in the cloud stacks and we'll want to keep it consistent for now, but it's a good quesiton to possibly revisit for all stacks, will check with @jarededwards and open a new issue if needed